modules/plottinggl/src/processors/persistencediagramplotprocessor.cpp

```diff
@@ -84,25 +100,40 @@ PersistenceDiagramPlotProcessor::PersistenceDiagramPlotProcessor()
 }
 
 void PersistenceDiagramPlotProcessor::process() {
-    if (brushing_.isConnected()) {
+    if (brushingPort_.isConnected()) {
+        if (brushingPort_.isChanged()) {
+            persistenceDiagramPlot_.setSelectedIndices(brushingPort_.getSelectedIndices());
+        }
+
         auto dataframe = dataFrame_.getData();
         auto dfSize = dataframe->getNumberOfRows();
 
         auto iCol = dataframe->getIndexColumn();
         auto &indexCol = iCol->getTypedBuffer()->getRAMRepresentation()->getDataContainer();
 
-        auto filteredIndicies = brushing_.getFilteredIndices();
+        auto filteredIndicies = brushingPort_.getFilteredIndices();
         IndexBuffer indicies;
         auto &vec = indicies.getEditableRAMRepresentation()->getDataContainer();
         vec.reserve(dfSize - filteredIndicies.size());
 
         auto seq = util::sequence<uint32_t>(0, static_cast<uint32_t>(dfSize), 1);
         std::copy_if(seq.begin(), seq.end(), std::back_inserter(vec),
```

Comment:
Idea for future: add BrushingAndLinking::getVisibleIndices  so that we do not need to do copy_if